pkg/operator/discover/discover.go

```diff
@@ -248,21 +255,21 @@ func (d *Discover) createDiscoverDaemonSet(ctx context.Context, namespace, disco
 
 }
 
+func fetchDiscoverSettings(key string, data map[string]string) string {
+	value := k8sutil.GetValue(data, key, "")
+	if value == "" {
+		value = os.Getenv(key)
```

Comment:
`k8sutil.GetValue()` already falls back to lookup the env var, so I don't think we need this helper method. Instead of `fetchDiscoverSettings()`, it should be the same to call `k8sutil.GetValue()`.

Reply:
I don't why added this method 😄